### PR TITLE
Roadmap

### LINKED ISSUE #210: Develop a Roadmap
Hello All, 

I'd like to help Dask-ML develop a roadmap, and @TomAugspurger requested that I open an issue here to kickstart the discussion. 

For those of you unfamiliar, a roadmap is a listing of near- and long-term goals that the project has, but which have yet to be implemented. These goals are generally larger than single pull request. "Add GPU Support" or "Port to Python 3" are examples of some roadmap items a project might have. However, goals can also include activities that seem small and mundane, but are critical to the project, such as "Improve Documentation" or "Achieve 100% Test Coverage."

**The purpose of this issue is so that roadmap items can be listed and commented on in the discussion.** Once we have a reasonable sense of what folks would like to see on the roadmap, the contents should make its way to the website/docs in some fashion. It is also a good idea if it ends up being listed in priority order, so that higher priority items are closer to the top. 

Some good example of project roadmaps are:

* [SymPy Roadmap](http://sympy.org/en/roadmap.html)
* [Bokeh Roadmap](https://bokehplots.com/pages/roadmap.html)

A decent example of a project roadmap is [Spyder's Roadmap](https://github.com/spyder-ide/spyder/wiki/Roadmap) which looks more like a timeline with milestones.

Roadmaps serve a joint purpose:

* They help inform and attract potentail funders, and some funders require roadmap documents
* They help attract contributors, as developers are more likely to jump in an help an establish project than start a new project to scratch their itch.

**TL;DR:** Roadmaps communicate a project's intentions. It would be a good idea for Dask-ML to have one, and I am happy to help facilitate!

So where would you like to see Dask-ML go?

This is a mirror issue to https://github.com/dask/dask/issues/3589

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes #210

### CODE REVIEW

**Comment by mrocklin** [comment]
Also, given that this sort of thing is likely to change frequently for a project like dask-ml, I wonder if it would be better suited as a living document like a google doc or hackmd page.

File: docs/source/roadmap.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+.. _roadmap:
+
+Dask-ML Roadmap
+===============
+
+Dask-ML wants to enable scalable machine learning in Python. It aims to do so by
+
+1. Working with existing libraries within the Python ecosystem
+2. Using the features of Dask to scale computation to larger datasets and larger
+   problems
```
Comment (stsievert) [inline docs/source/roadmap.rst]: > larger datasets and larger
   problems

Maybe "larger datasets and harder problems"?

File: docs/source/roadmap.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+We'd like to develop examples of working with large text datasets. Developing
+these examples might motivate some additional estimators and utilities that
+should be included in Dask-ML.
+
+Integrations with Other Frameworks
+----------------------------------
+
+Other Deep Learning and Machine Learning frameworks have their own distributed
+runtimes. Examples include Tensorflow, PyTorch, XGBoost, and LightGBM. Dask-ML
+is not interested in re-implementing everything these libraries too. Rather,
```
Comment (stsievert) [inline docs/source/roadmap.rst]: > everything these libraries too do

Typo: too => do

**Comment by stsievert** [comment]
Parts of this roadmap, or a summary of it, are featured on https://www.quansight.com/projects

> Dask ML has two goals. Scale machine learning to more computation; and scale machine learning to big data. Dask’s architecture can scale to many machines using parallelism for optimization.
​
> * Big Data Algorithms
> * Optimization Framework
> * Existing Libraries ​

with a big “coming soon” button at the bottom where other libraries link to their roadmap (e.g., https://docs.wixstatic.com/ugd/06679a_0a932ffb16e4445fba8bb0c7f8d81cd6.pdf). The two roadmaps I looked at were both PDFs styled the same way.

**Comment by scopatz** [comment]
@stsievert - those went up accidentally via an adminstrative error and were removed as soon as I discovered them. All apologies! 

We are currently waiting for this to be merged. Then we'll cut a draft of the brochure for your aproval and editting before anything hits our website.

**Comment by TomAugspurger** [comment]
On the call today, discussed a roadmap document that just lists the goals / values of Dask-ML, and links to Github issues for individual items. This seems like a good compromise for a project as young as dask-ml.

I've created a [roadmap label](https://github.com/dask/dask-ml/labels/Roadmap) and tagged a few issues with it.

**Comment by TomAugspurger** [comment]
CI failures are being tracked in #333 and fixed upstream.

**Comment by TomAugspurger** [comment]
Maybe I'll wait till Monday for a +1 from @stsievert or @mrocklin before merging. I hope this accurately captures what we discussed on the call.

**Comment by mrocklin** [comment]
+1 from me

**Pull request merged**